source/common/modules/markdown-editor/hooks/note-preview.js

```diff
@@ -0,0 +1,100 @@
+const tippy = require('tippy.js').default
+const { ipcRenderer } = require('electron')
+
+/**
+ * A hook for displaying link tooltips which can be used to visually
+ * enable users to click a link (without having to press down Ctrl/Cmd)
+ *
+ * @param   {CodeMirror}  cm  The instance to attach to
+ */
+module.exports = (cm) => {
+  cm.getWrapperElement().addEventListener('mousemove', (event) => {
+    let a = event.target
+
+    // Only for note links
+    if (!a.classList.contains('cm-zkn-link')) {
+      return
+    }
+
+    // If there's already a tippy on the instance, don't re-render it
+    if (a.hasOwnProperty('_tippy')) {
+      return
+    }
+
+    // Initialise displayed attributes to 'loading'
+    let title = 'loading'
+    let content = 'loading'
+    let wordCount = 'loading'
+    let time = 'loading'
+    // Create a tippy. This will display the loading values
+    let tooltip = tippy(a, {
+      content: 'Searching For File...',
+      allowHTML: true, // Obviously
+      interactive: true,
+      placement: 'top-start', // Display at the beginning of the anchor
+      appendTo: document.body, // anchor
+      showOnCreate: true, // Immediately show the tooltip
+      arrow: false, // No arrow for these tooltips
+      onHidden (instance) {
+        instance.destroy() // Destroy the tippy instance.
+      }
+    })
+
+    // Find the file's absolute path
+    ipcRenderer.invoke('application', { command: 'file-path-find', payload: a.innerText })
+      .then((filepath) => {
+        // If the file is found
+        if (filepath !== 'Not Found') {
+          // Retrieve the file contets
+          ipcRenderer.invoke('application', { command: 'get-file-contents', payload: filepath })
+            .then((descriptorWithContent) => {
+              // Remove html from content so html is displayed normally. (delete html tags)
+              descriptorWithContent.content = descriptorWithContent.content.replace(/(<([^>]+)>)/gi, '')
```

Comment:
Don't do this. I've recently added an HTML sanitizer that works fine for this. In general, I'd suggest we display a rendered HTML source (a.k.a. run the whole content through the `md2html`-source). But we might want to take a look at that, since something like this got me a CVE a few days ago, so … remove that line for now, but we must talk about this.